src/selectors.js

```diff
@@ -80,39 +84,68 @@ const defaultArr = [];
 
 const buildRecords = (ents = {}, key, opts = {}) => {
   if (!_.isObject(opts)) opts = {};
-  const { model, defaultJoins } = getModel(key);
+  const { defaultJoins } = getModel(key);
   const joinKeys = utils.composeArray(opts.joins, defaultJoins);
   let final = enrichRecords(ents, key) || defaultArr;
 
   joinKeys.forEach((k) => {
-    final = joins({
-      items: final,
-      mergeItems: enrichRecords(ents, k),
-      tableName: key,
-      mergeTableName: k,
-    });
+    if (typeof k === "object") {
+      const [mergeTableName, nestedMergeTableName] = Object.entries(k)[0];
+      final = joinsThrough({
+        items: final,
+        mergeItems: enrichRecords(ents, mergeTableName),
+        nestedMergeItems: enrichRecords(ents, nestedMergeTableName),
+        tableName: key,
+        mergeTableName: mergeTableName,
+        nestedMergeTableName: nestedMergeTableName,
+      });
+    } else {
+      final = joins({
+        items: final,
+        mergeItems: enrichRecords(ents, k),
+        tableName: key,
+        mergeTableName: k,
+      });
+    }
```

Comment:
Je zou hier nog `items` en `tableName` kunnen delen in `sharedProps`, is denk ik wel iets nicer, maar ook geen mega wins